R/deployApp.R

```diff
@@ -344,15 +354,6 @@ deployApp <- function(appDir = getwd(),
     pythonConfig <- pythonConfigurator(python, forceGeneratePythonEnvironment)
 
     logger("Inferring App mode and parameters")
-    appMetadata <- appMetadata(
-      appDir = appDir,
-      appFiles = appFiles,
-      appPrimaryDoc = appPrimaryDoc,
-      quarto = quarto,
-      contentCategory = contentCategory,
-      isCloudServer = isCloudServer,
-      metadata = metadata
-    )
```

Comment:
The `upload==FALSE` feels like a forgotten workflow. The amount of unnecessary work that is done when `upload==FALSE` surprises me; "redeploy without uploading" does not feel like it should interleave with the primary deploy workflow.

Should we make the separation between "upload" and "redeploy" more explicit?

Not a today task, but this code-movement made the additional work more obvious, since this call is moving out of the `upload==TRUE` path.

Reply:
What does deploy without upload mean?

Reply:
It looks like it only applies to shinyapps.io and ends up running the `deployApplication` against whatever we have stored as `bundle <- application$deployment$bundle`.

What this probably means is that rsconnect should run the "deployApplication" endpoint against whatever was the most recent bundle. I'm not sure about its history or if it still has a purpose.

@kippandrew, do you know if the `upload=FALSE` workflow (where the deploy happens without an upload) is valuable?

For Connect, the server supports such a thing, even though it's not made available today by rsconnect. If you call the "deploy" endpoint without a bundle identifier, Connect will "activate" the most recent bundle. That can be helpful if, for example, a systems change has been made that allows some package to successfully install. Typically, folks just upload again.